Parametric/Syntax/MTerm.agda

```diff
@@ -0,0 +1,192 @@
+  weaken-comp Γ₁≼Γ₂ (cForce x) = cForce (weaken-val Γ₁≼Γ₂ x)
+  weaken-comp Γ₁≼Γ₂ (cReturn v) = cReturn (weaken-val Γ₁≼Γ₂ v)
+  weaken-comp Γ₁≼Γ₂ (_into_ {σ} c c₁) = (weaken-comp Γ₁≼Γ₂ c) into (weaken-comp (keep σ •• Γ₁≼Γ₂) c₁)
+  weaken-comp Γ₁≼Γ₂ (cAbs {σ} c) = cAbs (weaken-comp (keep σ •• Γ₁≼Γ₂) c)
+  weaken-comp Γ₁≼Γ₂ (cApp s t) = cApp (weaken-comp Γ₁≼Γ₂ s) (weaken-val Γ₁≼Γ₂ t)
+
+  weaken-vals Γ₁≼Γ₂ ∅ = ∅
+  weaken-vals Γ₁≼Γ₂ (px • ts) = (weaken-val Γ₁≼Γ₂ px) • (weaken-vals Γ₁≼Γ₂ ts)
+
+  fromCBN : ∀ {Γ τ} (t : Term Γ τ) → Comp (fromCBNCtx Γ) (cbnToCompType τ)
+
+  fromCBNTerms : ∀ {Γ Σ} → Terms Γ Σ → Vals (fromCBNCtx Γ) (fromCBNCtx Σ)
+  fromCBNTerms ∅ = ∅
+  fromCBNTerms (px • ts) = vThunk (fromCBN px) • fromCBNTerms ts
+
+  open import UNDEFINED
+  -- This is really supposed to be part of the plugin interface.
+  cbnToCompConst : ∀ {Σ τ} → Const Σ τ → CompConst (fromCBNCtx Σ) (cbnToCompType τ)
+  cbnToCompConst = reveal UNDEFINED
+
+  fromCBN (const c args) = cConst (cbnToCompConst c) (fromCBNTerms args)
+  fromCBN (var x) = cForce (vVar (fromVar cbnToValType x))
+  fromCBN (app s t) = cApp (fromCBN s) (vThunk (fromCBN t))
+  fromCBN (abs t) = cAbs (fromCBN t)
+
+  -- To satisfy termination checking, we'd need to inline fromCBV and weaken: fromCBV needs to produce a term in a bigger context.
+  -- But let's ignore that.
+  {-# NO_TERMINATION_CHECK #-}
+  fromCBV : ∀ {Γ τ} (t : Term Γ τ) → Comp (fromCBVCtx Γ) (cbvToCompType τ)
+
+  -- This is really supposed to be part of the plugin interface.
+  cbvToCompConst : ∀ {Σ τ} → Const Σ τ → CompConst (fromCBVCtx Σ) (cbvToCompType τ)
+  cbvToCompConst = reveal UNDEFINED
+
+  cbvTermsToComps : ∀ {Γ Σ} → Terms Γ Σ → Comps (fromCBVCtx Γ) (fromCBVToCompList Σ)
+  cbvTermsToComps ∅ = ∅
+  cbvTermsToComps (px • ts) = fromCBV px • cbvTermsToComps ts
+
+
+  dequeValContexts : ValContext → ValContext → ValContext
+  dequeValContexts ∅ Γ = Γ
+  dequeValContexts (x • Σ) Γ = dequeValContexts Σ (x • Γ)
+
+  dequeValContexts≼≼ : ∀ Σ Γ → Γ ≼≼ dequeValContexts Σ Γ
+  dequeValContexts≼≼ ∅ Γ = ≼≼-refl
+  dequeValContexts≼≼ (x • Σ) Γ = ≼≼-trans (drop_••_ x ≼≼-refl) (dequeValContexts≼≼ Σ (x • Γ))
+
+  dequeContexts : Context → Context → ValContext
+  dequeContexts Σ Γ = dequeValContexts (fromCBVCtx Σ) (fromCBVCtx Γ)
+
+  fromCBVArg :
+    ∀ {σ Γ τ} → Term Γ τ →
+      (Val (cbvToValType τ • fromCBVCtx Γ) (cbvToValType τ) →
+        Comp (cbvToValType τ • fromCBVCtx Γ) (cbvToCompType σ)) →
+      Comp (fromCBVCtx Γ) (cbvToCompType σ)
+  fromCBVArg t k =
+    (fromCBV t) into
+      k (vVar vThis)
+
+  fromCBVArgs :
+    ∀ {Σ Γ τ} → Terms Γ Σ →
+      (Vals (dequeContexts Σ Γ) (fromCBVCtx Σ) →
+        Comp (dequeContexts Σ Γ) (cbvToCompType τ)) →
+      Comp (fromCBVCtx Γ) (cbvToCompType τ)
+  fromCBVArgs ∅ k = k ∅
+  fromCBVArgs {σ • Σ} {Γ} (t • ts) k =
+    fromCBVArg t (λ v →
+      fromCBVArgs
+        (weaken-terms (drop_•_ _ ≼-refl) ts) (λ vs →
```

Comment:
Here I'd like to write `drop _ • ≼-refl` like below, not `drop_•_ _ ≼-refl`, but Agda can't parse this. 
I remember the mixfix parsing paper discusses the limitations they have — @yfcai, do you understand why? What should we do to avoid the problem? Is the solution convenient enough?